shared/models/git_repo.rb

```diff
@@ -43,17 +43,21 @@ class GitRepo
     attr_accessor :repo_auth # whatever pieces of information that can change between git users
 
     attr_accessor :temporary_storage_path
+    # return [proc]
+    attr_accessor :callback
 
     class << self
       attr_accessor :git_action_queue
     end
 
     GitRepo.git_action_queue = TaskQueue::TaskQueue.new(name: "GitRepo task queue")
 
-    def initialize(git_config: nil, provider_credential: nil, async_start: false, sync_setup_timeout_seconds: 120)
+    def initialize(git_config: nil, provider_credential: nil, async_start: false, sync_setup_timeout_seconds: 120, callback: nil)
```

Comment:
I know this isn't added by this PR, but it would be good to have comments on what each of the parameters does here, e.g. `async_start` and `callback` 👍